internal/driver/mobile/canvas.go

```diff
@@ -10,124 +10,132 @@ import (
 	"fyne.io/fyne/v2/container"
 	"fyne.io/fyne/v2/driver/mobile"
 	"fyne.io/fyne/v2/internal/app"
-	"fyne.io/fyne/v2/internal/driver"
+	intdriver "fyne.io/fyne/v2/internal/driver"
 	"fyne.io/fyne/v2/internal/driver/common"
 	"fyne.io/fyne/v2/theme"
 	"fyne.io/fyne/v2/widget"
 )
 
-var _ fyne.Canvas = (*mobileCanvas)(nil)
+var _ fyne.Canvas = (*canvas)(nil)
 
-type mobileCanvas struct {
+type canvas struct {
 	common.Canvas
-
-	content          fyne.CanvasObject
-	windowHead, menu fyne.CanvasObject
-	scale            float32
-	size             fyne.Size
-
-	touched map[int]mobile.Touchable
-	padded  bool
-
-	onTypedRune func(rune)
-	onTypedKey  func(event *fyne.KeyEvent)
-
-	inited                bool
-	lastTapDown           map[int]time.Time
-	lastTapDownPos        map[int]fyne.Position
-	dragging              fyne.Draggable
-	dragStart, dragOffset fyne.Position
-
-	touchTapCount   int
+	content         fyne.CanvasObject
+	device          *device
+	dragOffset      fyne.Position
+	dragStart       fyne.Position
+	dragging        fyne.Draggable
+	initialized     bool
+	lastTapDown     map[int]time.Time
+	lastTapDownPos  map[int]fyne.Position
+	menu            fyne.CanvasObject
+	onTypedKey      func(event *fyne.KeyEvent)
+	onTypedRune     func(rune)
+	padded          bool
+	scale           float32
+	size            fyne.Size
 	touchCancelFunc context.CancelFunc
 	touchLastTapped fyne.CanvasObject
+	touchTapCount   int
+	touched         map[int]mobile.Touchable
+	windowHead      fyne.CanvasObject
 }
```

Comment:
The code was a lot easier to read when the fields were spaced out. Would you mind reverting that?

Reply:
Not if you give me a clear, unambiguous and exhaustive rule for the grouping.
And if you find such rule, give me a hint on how to do so. I tried for myself many times in my career and never was satisfied.
The best things I came up with were functionality groups covering a subset of all properties. In those cases, it turned out to be better to use a describing prefix for the props. This not only automates the grouping by sorting but also improves the readability in the rest of the code where a property is easier to be recognized as being part of a specific functionality.

Reply:
What do you mean? That doesn't change the fact that I think it would be a lot clearer code to have everything related to dragging in one group, touch related stuff in one; kind of like it was before.

Reply:
Maybe we can justify or keep this "unordered" groupings which I too find helpful by putting a comment at the top of each to communicate why it is a block - and keep each of them individually in order?

Reply:
If you think that the previous grouping had logic in it, you probably can name the groups, don’t you?
And of course the code is clearer if the properties that belong to the same functionality are grouped together. But this should better happen automatically when applying the code style rules, i.e., sorting.
The three dragging props are already grouped by their prefix.

Reply:
I could move these out of the “everything else” group like so:
```go
type canvas struct {
	common.Canvas
	content         fyne.CanvasObject
	device          *device
	initialized     bool
	lastTapDown     map[int]time.Time
	lastTapDownPos  map[int]fyne.Position
	menu            fyne.CanvasObject
	onTypedKey      func(event *fyne.KeyEvent)
	onTypedRune     func(rune)
	padded          bool
	scale           float32
	size            fyne.Size
	touched         map[int]mobile.Touchable
	windowHead      fyne.CanvasObject

	dragOffset      fyne.Position
	dragStart       fyne.Position
	dragging        fyne.Draggable

	touchCancelFunc context.CancelFunc
	touchLastTapped fyne.CanvasObject
	touchTapCount   int
}
```

Reply:
> Maybe we can justify or keep this "unordered" groupings which I too find helpful by putting a comment at the top of each to communicate why it is a block - and keep each of them individually in order?

Don’t get me wrong, I’m pro-grouping. But the former grouping was a mess. The only reasonable group was the `touch` group. And there were no comments.
IMO, comments are superfluous most of the time. Take `touchTapCount`: Its prefix clearly says where it belongs to. If you see this in the code, you immediately know, that it is about touch stuff. And if sorted alphabetically, it automagically groups with the other `touch*` properties. If you separate them from the others with blank lines, you don’t need a comment because the common prefix already says everything. If it would just be named `tapCount` you would loose these benefits.

Reply:
Sounds like we are on the same page then - grouping without need for comments.
Add the event handler fields in another group and what you have must be fairly close to what was there already?

Reply:
Not really. I the old grouping there was
```go
	touched map[int]mobile.Touchable
	padded  bool
```
which is totally unclear to me and
```go
	inited                bool
	lastTapDown           map[int]time.Time
	lastTapDownPos        map[int]fyne.Position
	dragging              fyne.Draggable
	dragStart, dragOffset fyne.Position
```
where I can only recognize the drag stuff.

Reply:
Maybe "fairly close" to me means different to you. I am happy with a logical tidy up, and some areas probably needed it - I do think we're on the same page.